check_hp_msa.py

```diff
@@ -305,43 +311,42 @@ def build_output(self):
                 health += " (%s)" % p['health-reason']
             if p['health-recommendation']:
                 health += " (%s)" % p['health-recommendation']
-            if p['error'] != 0:
-                health += " (error flag)"
-            if p['temperature-status'] != 'OK':
-                health += " - temperature %s %s" % (p['temperature-status'], p['temperature'])
-
-            self.output.append("[%s] %s %s %s %s %s %s" % (
-                p['location'].ljust(4, ' '),
-                p['vendor'],
-                p['model'],
-                p['size'],
-                p['serial-number'],
-                p['status'],
-                health,
-            ))
-
-            label = p['location'].replace('.', '_')
-
-            self.perfdata.append("disk_%s_temperature=%s" % (label, p['temperature-numeric']))
+            
+            # Handling output
+            outputLine=f"[{health}]" 
+            for output in self.outputList:
+                outputLine+=f" {p[output]}"
+            self.output.append(outputLine)
+           
+            # Handling PerfData
+            if self.perfDataList is not None:   
+                for perfDataLine in self.perfDataList:
+                    if isinstance(perfDataLine, str):
+                        self.perfdata.append(f"{self.objectName}_{perfDataLine}={p[perfDataLine]}")
```

Comment:
Perfdata label should be cleaned up and special chars replaced by `_`, so we have sth like `number_of_disks`